src/main/java/tdl/s3/upload/MultipartUploadFileUploadingStrategy.java

```diff
@@ -150,16 +158,23 @@ private void uploadRequiredParts(File file, String remotePath) throws IOExceptio
 
     private void submitUploadRequestStream(Stream<UploadPartRequest> requestStream) {
         requestStream.map(concurrentUploader::submitTaskForPartUploading)
-                .map(this::getUploadingResult)
+                .map(future -> {
+                    try {
+                        return this.getUploadingResult(future);
+                    } catch (DestinationOperationException ex) {
```

Comment:
I guess we should log this as error even if it's gracefully returns null for failure.